packages/solana/src/connectors/BaseConnector.ts

```diff
@@ -0,0 +1,392 @@
+  }
+
+  public get publicKey() {
+    return new PublicKey(SolStoreUtil.state.address ?? '')
+  }
+
+  protected async getProvider(): Promise<{
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    request: (args: any) => any
+  }> {
+    return Promise.reject(new Error('No provider in base connector'))
+  }
+
+  protected async constructTransaction<TransType extends keyof TransactionArgs>(
+    type: TransType,
+    params: TransactionArgs[TransType]['params']
+  ) {
+    const transaction = new Transaction()
+    const fromAddress = SolStoreUtil.state.address
+
+    if (!fromAddress) {
+      throw new Error('No address connected')
+    }
+    const fromPubkey = new PublicKey(fromAddress)
+
+    if (type === 'transfer') {
+      const transferParams = params as TransactionArgs['transfer']['params']
+      const toPubkey = new PublicKey(transferParams.to)
+      transaction.add(
+        SystemProgram.transfer({
+          fromPubkey,
+          toPubkey: new PublicKey(transferParams.to),
+          lamports: transferParams.amountInLamports
+        })
+      )
+      transaction.feePayer = transferParams.feePayer === 'from' ? fromPubkey : toPubkey
+    } else if (type === 'program') {
+      const programParams = params as TransactionArgs['program']['params']
+      transaction.add(
+        new TransactionInstruction({
+          keys: [
+            { pubkey: fromPubkey, isSigner: true, isWritable: programParams.isWritableSender }
+          ],
+          programId: new PublicKey(programParams.programId),
+          data: Buffer.from(base58.decode(JSON.stringify(programParams.data)))
+        })
+      )
+      transaction.feePayer = fromPubkey
+    } else {
+      throw new Error(`No transaction configuration for type ${String(type)}`)
+    }
+
+    const response = await this.requestCluster('getLatestBlockhash', [{}])
+
+    const { blockhash: recentBlockhash } = response.value
+    transaction.recentBlockhash = recentBlockhash
+
+    return transaction
+  }
+
+  protected async constructVersionedTransaction(params: TransactionArgs['transfer']['params']) {
+    const fromAddress = SolStoreUtil.state.address
+    if (!fromAddress) {
+      throw new Error('No address connected')
+    }
+    const fromPubkey = new PublicKey(fromAddress)
+    const toPubkey = new PublicKey(params.to)
+
+    const instructions = [
+      SystemProgram.transfer({
+        fromPubkey,
+        toPubkey,
+        lamports: params.amountInLamports
+      })
+    ]
+
+    const response = await this.requestCluster('getLatestBlockhash', [{}])
+    const { blockhash: recentBlockhash } = response.value
+
+    const messageV0 = new TransactionMessage({
+      payerKey: fromPubkey,
+      recentBlockhash,
+      instructions
+    }).compileToV0Message()
+
+    // Make a versioned transaction
+    const transactionV0 = new VersionedTransaction(messageV0)
+
+    return transactionV0
+  }
+
+  /*
+   * Public async sendTransaction(encodedTransaction: string) {
```

Comment:
@gooxpf @valentyn-solana Commented out, should we keep it?

Reply:
Leftover, will remove